Dockerfile

```diff
@@ -0,0 +1,20 @@
+FROM python:3.12.9
+
+WORKDIR /app
+
+# Copy only the essential files first for Docker layer caching.
+COPY pyproject.toml .
```

Comment:
why do you need this line? Won't `pyproject.toml` be copied over from `COPY . .`?